realsense2_camera/src/ros_sensor.cpp

```diff
@@ -102,6 +102,23 @@ void RosSensor::UpdateSequenceIdCallback()
     if (!supports(RS2_OPTION_SEQUENCE_ID))
         return;
 
+    bool is_hdr_enabled = static_cast<bool>(get_option(RS2_OPTION_HDR_ENABLED));
+
+    // Deleter to revert back the RS2_OPTION_HDR_ENABLED value at the end.
+    auto deleter_to_revert_hdr = std::unique_ptr<bool, std::function<void(void*)>>(nullptr,
```

Comment:
What if we pass the `is_hdr_enabled` to the deleter and add a condition inside?

Only turn it on if it was on.
This way we can avoid the reset.

What do you think?

Reply:
@Nir-Az ,
The unique_pointer is initialized with `nullptr`. So, by default, deleter will not be called.
If `is_hdr_enabled` is enabled alone, the unique_pointer is updated (at [line](https://github.com/IntelRealSense/realsense-ros/pull/2934/files#diff-440837bc6c77b72c935dc2c913ead13c628c014cbe0c150de364139a5c8736e3R119)).